packages/dds/tree/src/simple-tree/api/viewSchemaToSimpleSchema.ts

```diff
@@ -36,3 +46,88 @@ export function toSimpleTreeSchema(schema: ImplicitFieldSchema): SimpleTreeSchem
 		metadata: normalizedSchema.metadata,
 	};
 }
+
+/**
+ * Cache in which the results of {@link toSimpleNodeSchema} are saved.
+ */
+const simpleNodeSchemaCache = new WeakMap<TreeNodeSchema, SimpleNodeSchema>();
+
+/**
+ * Creates a {@link SimpleNodeSchema} from a {@link TreeNodeSchema}.
+ *
+ * @remarks Caches the result on the input schema for future calls.
+ */
+function toSimpleNodeSchema(schema: TreeNodeSchema): SimpleNodeSchema {
+	return getOrCreate(simpleNodeSchemaCache, schema, () => {
+		const kind = schema.kind;
+		switch (kind) {
+			case NodeKind.Leaf: {
+				return leafSchemaToSimpleSchema(schema);
+			}
+			case NodeKind.Map: {
+				return mapSchemaToSimpleSchema(schema);
+			}
+			case NodeKind.Array: {
+				return arraySchemaToSimpleSchema(schema);
+			}
+			case NodeKind.Object: {
+				assert(isObjectNodeSchema(schema), 0xa06 /* Expected object schema */);
+				return objectSchemaToSimpleSchema(schema);
+			}
+			default: {
+				unreachableCase(kind);
+			}
+		}
+	});
+}
+
+// TODO: Use a stronger type for leaf schemas once one is available (see object schema handler for an example).
```

Comment:
Not necessarily for this PR, but do we have `isArrayNodeSchema`, etc. helpers now that we could leverage to improve type-safety here?

Reply:
Ya, I think it makes sense to do that when we make the change to add ArrayNodeSchema though.